### ISSUE TITLE
ABI Method - Automatic TX Type Checking Makes Sense

### ISSUE THREAD

**Issue by mangoplane** [FrankSzendzielarz/AlgorandVisualStudio#27]
For ABI methods overloaded with a *particular type of TX reference*, say `AssetTransferTransactionReference`, I believe it should unmarshall to handle the type checking automatically. I really can't see a reason why this flexibility might not be desired. Maybe the default transaction reference, `TransactionReference` can be a bare bones flexible transaction handler for a case where the dev wants to break out of the flow.

```
              string txTypeCheck = "axfer";
            if (txn.TxType != txTypeCheck.ToByteArray()) return 0;
```

Kind regards.

**Comment by FrankSzendzielarz** [comment]
Agreed. This functionality is not there simply because it wasnt in the roadmap prior and wasnt sure if the preference would be to have the plumbing do it for us or not. Will tag as feature request
